### PR TITLE
Road name spelling consistency check

### PR DESCRIPTION
**Pull request by seancoulter**
### Description:
This is a new check that's designed to flag road segments whose Name tags have slightly different spellings from one another. "Slightly different spelling" is understood to mean a edit distance of 1 between two strings. An edit distance is defined to be the number of character insertions, deletions, and substitutions needed for two strings to exactly match. The program collects all road segments in a configurable search area then filters them for those that have slightly different spellings. The goal is to bring attention to areas of a road with spelling mixups so that editors can create a uniform spelling for the entire road.

This check should flag road segments whose Name tags contain:
--Accented character inconsistencies
--Possessive apostrophe misplacement/absence
--Other character additions/deletions

### Potential Impact:

None. This is a new check with no dependencies.

### Unit Test Approach:

Unit test 

### Test Results:

Unit tests pass

### CODE REVIEW

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    @SuppressWarnings("3655")
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+
+        final Edge edge = (Edge) object;
+
+        // If this edge's NameTag doesn't exist, it's skipped
+        if (!edge.getName().isPresent())
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java]: It looks like we are already checking for this in validCheckForObject.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+    /**
+     * @param startEdge
+     *            the edge from which the search started
+     * @param maximumAllowedDifferences
+     *            the number of Levenshtein edits allowed
+     * @return edges that are in the search area and whose names are at most the desired Levenshtein
+     *         distance from the start edge's
+     */
+    static Predicate<Edge> getCandidateEdges(final Edge startEdge,
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: The name of this function doesn't make a lot of sense, as it does not return Edges.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+            final int maximumAllowedDifferences)
+    {
+        // evaluate name tag of startEdge vs incoming Edge
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: This can go in the java doc.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+     * Compute the Levenshtein distance between two road names. Code used to calculate Levenshtein
+     * distance is adapted from https://www.baeldung.com/java-levenshtein-distance
+     *
+     * @param incomingEdgeName
+     *            the name of the next edge in the search area
+     * @param startingEdgeName
+     *            the name of the edge from which the search started
+     * @return the Levenshtein distance between two edge's names
+     */
+    private static int getLevenshteinDistance(final String incomingEdgeName,
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: There is an existing implementation of this available through on of the apache libraries: `org.apache.commons.lang3.StringUtils.getLevenshteinDistance()`. I'm not sure how useful it is given the need for handling the direction and number differences.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+     * @param maximumSearchDistance
+     *            the maximum distance from the end of the incoming edge to the start of the
+     *            starting edge
+     * @return A stream of edges that fall in the search area
+     */
+    private static Function<Edge, Stream<Edge>> edgesWithinMaximumSearchDistance(
+            final Edge startEdge, final Distance maximumSearchDistance)
+    {
+        return incomingEdge -> incomingEdge.end().getLocation()
+                .distanceTo(startEdge.start().getLocation())
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java:134]: It seems a bit odd to me to use an EdgeWalker to try to find things in an area. It would make more sense to use linear or network distance with an EdgeWalker, I think. If for this check it makes sense to use area, then we should be using the atlas spatial indices and do something like `atlas.edgesWithin(start.bounds().expand(maximumSearchDistance))`

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        final int[][] results = new int[incomingEdgeName.length() + 1][startingEdgeName.length()
+                + 1];
+
+        // Roads differ by one directional character (N,S,E, or W) in their name strings
+        boolean possibleDirectionalDifference = false;
+
+        // Roads differ by one number character (1-9) in their name strings
+        boolean singleNumericalDifference = false;
+
+        for (int i = 0; i <= incomingEdgeName.length(); i++)
```
Comment (Bentleysb) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: i -> index
Abbreviations aren't permitted in this project.

**Review by Bentleysb — CHANGES_REQUESTED** [review]
This looks like a good start. In addition to my other comments, it looks like you need to add descriptions to a lot of your java docs.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java (hunk 4 above)
Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: Right, the main problem with that implementation is it doesn't handle direction and number differences. We need more granular control of the Levenshtein function to work with direction/number differences so I decided a custom Levenshtein method would do best. Also, ```StringUtils.getLevenshteinDistance()``` is deprecated.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java (hunk 2 above)
Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: how about ```isEdgeWithInconsistentSpelling()```?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java (hunk 5 above)
Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java:134]: Thanks for the suggestion--I didn't know about this approach. I think it's a great idea, but before I make any major changes to the search logic I'd like to point out one thing. The current EdgeWalker approach ensures that collected edges are part of the same road network, meaning we can guarantee that all collected edges have a common connection in the ```startEdge```. Using spatial indices seems like it raises the possibility of collecting edges that aren't connected to the ```startEdge```, or are only connected through a ton of degrees of separation, which I believe means they probably aren't from the same road.

Do you have any thoughts about this?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * Flags road segments that have no {@link org.openstreetmap.atlas.tags.names.NameTag} or a NameTag
```
Comment (sayas01) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java]: nit: you can import the NameTag here.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+
+    private static final int NINE = 9;
+
+    private static final int ZERO = 0;
+
+    // ASCII character to digit offset
+    private static final int ASCII_OFFSET = 48;
+
+    /**
+     * Evaluate the {@link org.openstreetmap.atlas.tags.names.NameTag}s of the startingEdge and an
```
Comment (sayas01) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: nit: static import

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge
+                && ((Edge) object).highwayTag().isMoreImportantThanOrEqualTo(
+                        MINIMUM_NAME_PRIORITY_DEFAULT)
+                && ((Edge) object).isMasterEdge() && !this.isFlagged(object.getIdentifier())
+                && ((Edge) object).getName().isPresent();
```
Comment (sayas01) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java:59]: Is the requirement only to correct NameTag? Could this be extended to other name tags like alt_name etc?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+     *
+     * @param incomingEdge
+     *            the next edge in the search area
+     * @param startEdge
+     *            the edge from which the search started
+     * @return the Levenshtein distance between two edge's names, or -1 if the incoming edge doesn't
+     *         have a name, or 0 if the names are the same
+     */
+    @SuppressWarnings("squid:S3655")
+    private static int similarityIndex(final Edge incomingEdge, final Edge startEdge)
```
Comment (sayas01) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: Since this gives the similarity index of two strings, I think we can just pass the name tags instead of the edge itself.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java (hunk 5 above)
Comment (Bentleysb) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java:134]: I see what you mean, that does make sense. It still feels a bit weird, but I don't have a better solution at the moment.

**Review by sayas01 — COMMENTED** [review]
Looks like the config file and md document are missing from the PR.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+     *            starting edge
+     * @return A stream of edges that fall in the search area
+     */
+    private static Function<Edge, Stream<Edge>> edgesWithinMaximumSearchDistance(
+            final Edge startEdge, final Distance maximumSearchDistance)
+    {
+        return incomingEdge -> incomingEdge.end().getLocation()
+                .distanceTo(startEdge.start().getLocation())
+                .isLessThanOrEqualTo(maximumSearchDistance)
+                        ? incomingEdge.connectedEdges().stream().filter(Edge::isMasterEdge)
```
Comment (sayas01) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java:136]: Do we need connected edges with no name tags? If not, we can filter them here.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java (hunk 9 above)
Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java:59]: The requirements are self-imposed so far, so we could definitely look at the other tags.I figured we'd only want to flag the most "visible" (e.g. can see in the editor) names, but I'm open to your idea. Maybe it could be a configurable list of name related tags to flag

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+ */
+public class RoadNameSpellingConsistencyCheck extends BaseCheck<Long>
+{
+
+    private static final long serialVersionUID = 1L;
+
+    private static final HighwayTag MINIMUM_NAME_PRIORITY_DEFAULT = HighwayTag.SERVICE;
+    private static final double MAXIMUM_SEARCH_DISTANCE_DEFAULT = 300.0;
+    private static final double MAXIMUM_CHARACTER_DIFFERENCES_DEFAULT = 1;
+    private static final String FALLBACK_INSTRUCTIONS = "These road segments have spelling inconsistencies. One spelling is: %s. Examine all flagged road segments to determine the best spelling, and apply this spelling to all of those segments.";
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java]: nit: I think instead of `%s` you can just do `{}` and let the CheckFlag build the string for you

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java (hunk 9 above)
Comment (nachtm) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java:59]: I'm a big fan of potentially looking at other names!

One possible concern with this, though, is that edges can have multiple names, which might differ by one character in valid ways. For each rule I can come up to handle this, I can come up with a counterexample where that rule doesn't make sense. Maybe there's something I haven't thought of, though :)

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 13]
```diff
@@ -0,0 +1,299 @@
...
+    }
+
+    /**
+     * Check two characters to see if they're equal.
+     *
+     * @param evaluateEdgeCharacter
+     *            the character being checked for equality with the lambda parameter
+     * @return true if the incoming character is the same as the parameter character
+     */
+    private static Predicate<String> equalCharacter(final char evaluateEdgeCharacter)
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: The predicate that this returns is only looking at the first character of the string. Could you clarify that in the comments?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java (hunk 4 above)
Comment (nachtm) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: First of all, respect for this! Love to see dynamic programming in the wild 🙌

I would love to see this pulled out into a utility somewhere. In addition to making it public for others to use, that allows us to write unit tests for it which will increase our confidence in this covering all of the edge cases.

As another thought, do you see the threshold for this ever increasing above 1? Or does it not make sense to ever check for a minimum edit distance of 2, for example?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+                                            startingEdgeName.charAt(startingEdgeNameIndex - 1)),
+                            results[incomingEdgeNameIndex - 1][startingEdgeNameIndex] + 1,
+                            results[incomingEdgeNameIndex][startingEdgeNameIndex - 1] + 1);
+                }
+            }
+        }
+        // If there's only a single character difference and that character is a directional OR
+        // numerical character, we consider both roads to be different and so we don't flag them.
+        // Else we return the Levenshtein distance as usual.
+        return (possibleDirectionalDifference || singleNumericalDifference)
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: One way to make this more general for other users would be to return the actual edit operations necessary to get from one string to another. In our case, we can search those operations for a directional difference or a numerical difference and update as appropriate. In other cases, a user could take the raw levenshtein distance by counting the number of operations, etc. [Take a look at this stackoverflow post for an example of what I'm talking about, along with sample implementations](https://stackoverflow.com/questions/10638597/minimum-edit-distance-reconstruction/10639697#10639697).

If you took that step, I think this would be very useful outside of this class.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 15]
```diff
@@ -0,0 +1,299 @@
...
+     *            the incoming Edge's character
+     * @param startingEdgeCharacter
+     *            the starting Edge's character
+     * @return true if the parameter characters are both directional (members of Direction enum) AND
+     *         they are different from one another, false otherwise
+     */
+    private static boolean hasDirectionalCharacterDifference(final char incomingEdgeCharacter,
+            final char startingEdgeCharacter)
+    {
+        return DIRECTIONS.stream().map(Enum::toString).filter(
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: How about [ApacheCommons](https://stackoverflow.com/a/21730297)?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 16]
```diff
@@ -0,0 +1,299 @@
...
+     *            the incoming Edge's character
+     * @param startingEdgeCharacter
+     *            the starting Edge's character
+     * @return true if both parameter characters are different numbers in [0,9], false otherwise
+     */
+    private static boolean hasNumericalCharacterDifference(final char incomingEdgeCharacter,
+            final char startingEdgeCharacter)
+    {
+        return Character.isDigit(incomingEdgeCharacter) && Character.isDigit(startingEdgeCharacter)
+                && IntStream.range(ZERO, NINE)
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: This is clever! However, since we already know that both are digits, couldn't we just do `incomingEdgeCharacter != startingEdgeCharacter`?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java [hunk 17]
```diff
@@ -0,0 +1,299 @@
...
+    /**
+     * Retrieve the minimum value out of parameter numbers.
+     *
+     * @param numbers
+     *            the numbers on which to operate
+     * @return the minimum of those numbers
+     */
+    private static int min(final int... numbers)
+    {
+        return Arrays.stream(numbers).min().orElse(Integer.MAX_VALUE);
```
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: I know we're pulling in at least part of the guava library already. If we're already pulling in guava's `Ints` class (I don't know for sure if we are or not), we could replace all of the calls to this method with [Ints.min()](https://guava.dev/releases/snapshot/api/docs/com/google/common/primitives/Ints.html#min-int...-)

**Review by nachtm — COMMENTED** [review]
Cool cool cool cool! 

A couple comments but I like the approach :)

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java (hunk 9 above)
Comment (nachtm) [inline src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java:59]: I commented this on @Bentleysb's comment, but it's already been resolved so it's hidden. Take two:
First of all, respect for this! Love to see dynamic programming in the wild 🙌

I would love to see this pulled out into a utility somewhere. In addition to making it public for others to use, that allows us to write unit tests for it which will increase our confidence in this covering all of the edge cases.

As another thought, do you see the threshold for this ever increasing above 1? Or does it not make sense to ever check for a minimum edit distance of 2, for example?

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java (hunk 12 above)
Comment (danielduhh) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java]: Can you add both road names to the instructions so it's easy to compare?

Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java]: I can do that. It works now by collecting all the road names that are spelled inconsistently with respect to the starting edge's name, so it'd be a (potentially long) list of names in the instruction. Not sure if we want to keep it under a character limit though

Comment (danielduhh) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java]: What if you made the list unique ?

Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java]: That sounds like a great place to start

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java (hunk 9 above)
Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheck.java:59]: Update:
From the discussion with the devs a week or so ago, I gathered that it's not the right time to pull this check out into a utility, mainly because we're only looking for edit distances of 1. I'd think we'd really only want a utility if we were looking at variable edit distances.

I like the idea of this being a utility. I can see it being useful for at least a few cases when we're dealing with tags. I'd have to do some research and then I think we can get the discussion going on how exactly we want to implement it.

Regarding your question, I don't see the threshold increasing above 1. The task for this problem is to catch "smaller" issues in spelling (accent mixup, one letter missing/added, etc) which I believe are all covered by an edit distance of 1. However I do think we should allow the edit distance to be bumped up if we pull out a utility. If we decide to go that route, I suggest we get deeper test coverage, since an increased edit distance brings substantially more corner cases/things to watch out for.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java (hunk 14 above)
Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java]: Thanks. I've decided to take a more general approach to numerical/directional differences by wrapping them in what I refer to as Identifier Strings. These Identifiers are meant to handle road names whose numbers/directions differ by more than a single character.

I do think we should track all edit operations for our utility, though. I can see it being useful if we wanted to dig into patterns of TPs vs FPs.

File: src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java (hunk 11 above)
Comment (seancoulter) [reply src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java:136]: Made that change, but reverted it because it was dropping some TPs. I'm a bit surprised by that since it looks like it's only a difference in _where_ EdgeWalker filters these Edges without names.

**Comment by seancoulter** [comment]
This should be ready for another round of reviews. Most of the changes are in the getLevenshteinDistance() function. Specifically, how the program decides not to flag road segments that have different Identifier Strings (defined by ALPHANUMERIC_IDENTIFIER_STRING_REGEX and CHARACTER_IDENTIFIER_STRING_REGEX in the Walker class).

**Comment by seancoulter** [comment]
Thanks everyone for the reviews. I think I've addressed all of your comments. I'm hoping to close this out soon, so if you have anything else you might want to add, please feel free!

@Bentleysb @sayas01 @nachtm

File: config/configuration.json [hunk 18]
```diff
@@ -267,6 +267,9 @@
       "instruction": "Correct the areas as necessary to not have overlapping ways representing the same AOI",
       "difficulty": "HARD"
     }
+  },
+  "RoadNameSpellingConsistencyCheck": {
```
Comment (sayas01) [inline config/configuration.json:271]: We should also add MR info.
